metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/OrdinalFunctions.java

```diff
@@ -37,17 +35,23 @@
    * Return the maximum value of a list of input values in a Stellar list
    */
   @Stellar(name = "MAX"
-          , description = "Returns the maximum value of a list of input values"
-          , params = {"list - List of arguments. The list may only contain objects that are mutually comparable / ordinal (implement java.lang.Comparable interface)" +
+          , description = "Returns the maximum value of a list of input values or from a statistics object"
+          , params = {"stats - The Stellar statistics object"
+          ,"list - List of arguments. The list may only contain objects that are mutually comparable / ordinal (implement java.lang.Comparable interface)" +
           " Multi type numeric comparisons are supported: MAX([10,15L,15.3]) would return 15.3, but MAX(['23',25]) will fail and return null as strings and numbers can't be compared."}
-          , returns = "The maximum value in the list, or null if the list is empty or the input values were not comparable.")
+          , returns = "The maximum value in the list or from stats, or null if the list is empty or the input values were not comparable.")
   public static class Max extends BaseStellarFunction {
 
     @Override
     public Object apply(List<Object> args) {
       if (args.size() < 1 || args.get(0) == null) {
         throw new IllegalStateException("MAX function requires at least a Stellar list of values");
       }
+      Object firstArg = args.get(0);
+      if(firstArg instanceof Ordinal) {
+        Ordinal stats = convert(firstArg, Ordinal.class);
+        return stats.getMax();
+      }
       Iterable list = (Iterable<Object>) args.get(0);
```

Comment:
It would make sense to wrap the existing "iterable" handling code in an "else if".  And also handle the possibility that the argument is not an Iterable nor an Ordinal. Perhaps like so...

```
Object firstArg = args.get(0);
if(firstArg instanceof Ordinal) {
  <your new code>

} else if(firstArg instanceof Iterable) {
  <existing code>

} else {
   throw new IllegalStateException("MAX function expects either .... ");

}
```